### ISSUE TITLE
Using "$ref" and "oneOf" occurs JSONSchemer::InvalidFileURI  

### ISSUE THREAD

**Issue by devleoper** [davishmcclurg/json_schemer#43]
Somehow I got an issue on both latest release and master branch for those cases.

### root.json
```json
{
  "type": "object",
  "properties": {
    "foo": {
      "$ref": "foo.json"
    }
  }
}
```

### foo.json
```json
{
  "definitions": {
    "foo1": {
      "type": "object",
      "properties": {
        "name": { "type": "string" },
        "bar": { "$ref": "bar.json" }
      }
    },
    "foo2": {
      "type": "object",
      "properties": {
        "description": { "type": "string" }
      }
    }
  },
  "allOf": [
    { "$ref": "#/definitions/foo1" },
    { "$ref": "#/definitions/foo2" }
  ]
}
```

### bar.json
```json
{
  "type": "object",
  "properties": {
    "id": { "type": "number" }
  }
}
```

and this causes this example into `JSONSchemer::InvalidFileURI` error which doesn't make sense.

```ruby
require "json_schemer"
schema = JSONSchemer.schema(Pathname.new("path/to/root.json"))
schema.valid?(
  "foo" => {
    "name" => "Greeter",
    "bar" => { "id" => 123 },
    "description" => "Hello World!"
  }
) # => must use `file` scheme (JSONSchemer::InvalidFileURI)
```

while this example works fine.

```ruby
require "json_schemer"
schema = JSONSchemer.schema(Pathname.new("path/to/foo.json"))
schema.valid?(
  "name" => "name",
  "bar" => { "id" => 123 },
  "description" => "Hello World!"
) # => true
```

Looking through the stack trace `URI` for `bar.json` became `#<URI::Generic bar.json>` in the `$ref`-resolving steps.

**Comment by davishmcclurg** [comment]
Thanks for the detailed issue! I was able to reproduce it. I've tracked it down to an issue setting `parent_uri` for ref schemas.

**Comment by davishmcclurg** [comment]
I released a fix (47d40f39f873e109443986fcfb4e5f61b0387087) in version 0.2.7.
